### ISSUE TITLE
Using `--spec` option results in "no specs found" when project is located at Docker root directory

### ISSUE THREAD

**Issue by BenDaSpur** [cypress-io/cypress#23380]
### Current behavior

![image](https://user-images.githubusercontent.com/2095474/184974959-2ceec018-f094-45d2-acb5-c560d68170d8.png)

All that changed is the version number in my dockerfile

If I downgrade from there to 10.4.0 I have no issues.

### Desired behavior

Tests should run with no errors

### Test code to reproduce

Build cypress/included:10.5.0

Try doing a `cypress run`

### Cypress Version

10.5.0

### Node version

16.16.0

### Operating System

Linux

### Debug Logs

```shell
Your supportFile is missing or invalid: support/e2e.ts

The supportFile must be a .js, .ts, .coffee file or be supported by your preprocessor plugin (if configured).

Fix your support file, or set supportFile to false if a support file is not necessary for your project.

If you have just renamed the extension of your supportFile, restart Cypress.

https://on.cypress.io/support-file-missing-or-invalid
```

**Comment by tbiethman** [comment]
@BenDaSpur this issue was [previously reported](https://github.com/cypress-io/cypress/issues/23361) and fixed in 10.6.0, which was just released. Can you try 10.6.0 and report back?

**Comment by BenDaSpur** [comment]
@tbiethman I am not having that issue anymore but on 10.6 I am getting this issue saying no spec files were found
![image](https://user-images.githubusercontent.com/2095474/184986105-fde14c8d-b412-4caa-950d-6a5eaaa35370.png)

and in my config I have:
`specPattern: 'cypress/e2e/tests/**/*.cy.ts',`

**Comment by tbiethman** [comment]
@BenDaSpur hmm, are you running your command with a particular `--spec` filter?

**Comment by BenDaSpur** [comment]
@tbiethman yes, this is what it looks like
`cypress run --spec ./cypress/e2e/tests/api/REST.cy.ts --headless --browser chrome --config chromeWebSecurity=false,trashAssetsBeforeRuns=false,baseUrl=https://demo.wavv.com --reporter mochawesome --reporter-options reportFilename=REST,html=false,overwrite=false,reportDir=/cypress/results/tests --env SERVER=***,SERVER_URL=https://*********.com,VENDOR_ID=********,API_KEY=******`

**Comment by tbiethman** [comment]
@BenDaSpur gotcha, that looks right. Can you share how you're using the container in a little more depth? It looks like you have your project files/are running cypress from the root directory?

**Comment by BenDaSpur** [comment]
```
FROM cypress/included:10.6.0 as base
ENV NODE_ENV=production
# RUN apt-get update
WORKDIR /
COPY ./package.json ./package-lock.json ./

ENV CI=1
RUN npm ci --only=production
RUN cypress verify

FROM base as target
COPY ./cypress.config.ts ./
COPY ./cypress ./cypress
COPY ./entrypoint.sh ./
COPY ./tsconfig.json ./

ENTRYPOINT [ "/entrypoint.sh" ]
```

My entrypoint script is where that cypress run is happening.

@tbiethman Let me know if you need more info

**Comment by tbiethman** [comment]
@BenDaSpur thank you very much. I can reproduce this with a similar Dockerfile within our [reproduction template repo](https://github.com/cypress-io/cypress-test-tiny):

```
FROM cypress/included:10.6.0 as base
# RUN apt-get update
WORKDIR /
COPY ./package.json ./package-lock.json ./

RUN npm i
RUN cypress verify

FROM base as target
COPY ./cypress.config.js ./
COPY ./cypress ./cypress

ENV DEBUG=cypress:data-context:sources:FileDataSource

ENTRYPOINT [ "cypress", "run", "--spec", "**/*.js" ]
```

The issue is that we're not properly handing projects at root with 10.5.0/10.6.0. You could work around this in the meantime by updating your dockerfile to copy your project into a nested directory. But this is a bug and it will get fixed.
